parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -33,8 +33,10 @@ if [[ "${TARGET_ENVIRONMENT,,}" == "${AZURE_STACK_ENV}"  ]]; then
     config_script_custom_cloud=/opt/azure/containers/provision_configs_custom_cloud.sh
     wait_for_file 3600 1 $config_script_custom_cloud || exit $ERR_FILE_WATCH_TIMEOUT
     source $config_script_custom_cloud
+    CUSTOM_CLOUD_SUFFIX=${AZURE_STACK_SUFFIX}
 fi
 
+KUBERNETES_VERSION_SUFFIX="${CUSTOM_CLOUD_SUFFIX:-$AZURE_SUFFIX}"
```

Comment:
instead of using empty variables, can we do:

```
if [[ "${TARGET_ENVIRONMENT,,}" == "${AZURE_STACK_ENV}"  ]]; then 
    KUBERNETES_VERSION=${KUBERNETES_VERSION}${AZURE_STACK_SUFFIX}
fi
```

?

Reply:
Maybe, but I am a little hesitant to override `KUBERNETES_VERSION` in this case.

The reason is that I fear someone in the future my break Azure Stack's deployments by adding a feature that assumes the `X.X.X` format for `KUBERNETES_VERSION`. But I understand why you want that inside the `if` statement.

Example of this could be `cleanUpContainerImages`. I looks like it won't correctly if we add the suffix?

Reply:
Thought about this on my way home. My concern is missing the point as we will only offer validated VHDs. Will update.